internal/config/database.go

```diff
@@ -77,12 +88,11 @@ func (d *Database) Open(logger *logging.Logger) (*icingadb.DB, error) {
 			return nil, errors.Wrap(err, "can't open mysql database")
 		}
 
-		wsrepSyncWait := int64(d.Options.WsrepSyncWait)
-		setWsrepSyncWait := func(ctx context.Context, conn driver.Conn) error {
-			return setGaleraOpts(ctx, conn, wsrepSyncWait)
+		connectorCallbacks.OnInitConn = func(ctx context.Context, conn driver.Conn) error {
+			return setGaleraOpts(ctx, conn, int64(d.Options.WsrepSyncWait))
```

Comment:
We use almost all options concurrently. Though it makes sense to make `Options` private.

Reply:
> Though it makes sense to make Options private.

By "private" do you mean unexported? If so, the Yaml parser will not be able to decode config into it and usages like `db.Options.MaxPlaceholdersPerStatement` outside the database package are not going to work as well.

Reply:
> If so, the Yaml parser will not be able to decode config into it

I did not refer to the `Database#Config` struct. The YAML parser would not be a problem.

> and usages like `db.Options.MaxPlaceholdersPerStatement` outside the database package are not going to work as well.

That's right, and that also applies to the Redis options. So that's not the solution. But again, to be honest, I wouldn't worry too much about using options concurrently, as we basically do this for every option.